HelpSource/Guides/WritingClasses.schelp

```diff
@@ -106,101 +186,176 @@ we are calling the code::.init:: method on that object, which is an instance met
+	}
+}
+::
+
+
+
+
+section::Alternatives to inheritance
+
+Overreliance on inheritance is usually a design flaw. Inheritance is mainly a way to organise code, and shouldn't be mistaken for a categorisation of objects. Two objects may respond to a message in different ways (polymorphism), and objects delegate control to ther objects they hold in their instance variables (object composition).
+
+subsection::Polymorphism
+
+See also: link::Guides/Polymorphism::
+
+Two completely unrelated objects can respond to the same messages and therefore be used together in the same code. For example, link::Classes/Function:: and link::Classes/Event:: have no common superclass apart from the general class link::Classes/Object::. But both respond to the message code::play::. Instead of inheriting all methods, you can simply implement some of the same methods in your class.
 
 code::
-Superclass {
-    calculate { arg in; in * 2 }
-    value { arg in; ^this.calculate(in) }
+MyClass {
+	var count = 0;
+	value {
+		^count = count + 1
+	}
 }
+// objects of this class will respond to the message "value", just like a function.
+a = MyClass.new;
+a.value; // returns 1
+::
+
 
-MyClass : Superclass {
-    calculate { arg in; in * 3 }
+subsection::Object Composition
+Often, an object passes control to one of the objects it has in its instance variables. Because these objects can be of any kind, this is a very flexible way to achieve a wide range of functionalities. For example, a link::Classes/Button:: has an code::action:: instance variable, which may hold anything that responds to the message code::value::.
+
+code::
+MyClass {
+	var action;
+	*new { |action|
+		^super.newCopyArgs(action)
+	}
+	value {
+		action.value;
+	}
 }
+
+// depending on what "action" is, objects of this class will behave differently
+a = MyClass({ "hello." });
+b = MyClass({ |i| log2(i) * sin(i * pi) });
+a.value(8);
+b.value(8);
 ::
 
-If the method of the superclass is needed, it can be called by super.
+Often, variables like code::action:: above are filled with custom objects that belong to code::MyClass::. Thus, one will write many small classes that can be well combined in such a way. This is called "pluggable behavior".
 
-code::
-Superclass {
-    var x;
 
-    init {
-        x = 5;
-    }
+section::Variables
+
+subsection:: Initializing variables directly
+
+In a variable declaration, variables can be directly initialized. This works for link::Reference/Literals:: only.
+
+code::
+MyClass {
+	classvar all = #[];
+	var x = 8;
+	var y = #[1, 2, 3];
 }
+::
+
+In class definitions, it is not possible to chain assignments (e.g. code:: var x = 9; var y = x + 1::).
+
+
+subsection:: Variable Scope
+
+An instance variable is accessible from all instance methods of this class and its subclasses. A class variable is accessible from all class and instance methods of this class and its subclasses. Instance variables of the same name as override the scope of class variables of the same name.
```

Comment:
The first sentence here is duplicated. :) Also, I think the second sentence has a grammar mistake, maybe it should be "Instance variables will shadow class variables of the same name".